core/src/main/scala/knobs/AWS.scala

```diff
@@ -36,59 +37,67 @@ object aws {
 
   private val defaultFormating: String => String = x => "\"" +x+ "\""
 
+  private def or[A](x: IO[A], y: => IO[A]): IO[A] =
+    x.attempt.flatMap {
+      case Left(_)  => y
+      case Right(a) => IO.pure(a)
+    }
+
   /**
    * attempt to read an item of metadata from the AWS metadata service in under 300ms.
    */
-  private def fetch(child: Path, parent: Path = root, version: Path = revision): Task[String] =
-    Task {
+  private def fetch(child: Path, parent: Path = root, version: Path = revision): IO[String] =
+    IO {
```

Comment:
This does blocking IO.  It used to thread shift, but I don't know that it did _usefully_.

Reply:
Right, given the old behavior of relying on the magically default `Strategy` and my old habit of confusing `Task.apply` with `Task.delay` I opted to replace all use cases with `IO.apply` unless a thread pool was passed in explicitly, which it was in one or two places. I can go back and change all these to `shift *> apply` if that is preferred.

Reply:
I think this is okay.